arrow-fx-coroutines/README.MD

```diff
@@ -0,0 +1,480 @@
+
+`parTraverse` can be used to traverse any data type that supports `Traverse` such as `List`, `Validated`, `Either`, `Option`, etc.
+It is used to map elements of the same type `A` in parallel. Cancelling the caller `suspend fun` will cancel all running `Fiber`s inside `parTraverse` gracefully. 
+
+### RaceN
+
+`raceN` is similar to `parMapN` except for it only returns the winner of the race, and cancels the loser as soon as the first fiber finishes the race.
+
+#### RacePair/RaceTriple/ForkAndForget
+
+There are a couple of low level constructs available, and they should be used with care!
+
+- racePair/raceTriple like `raceN` races 2 operations, but instead of cancelling the loser it returns the `Fiber` of the loser task
+
+- ForkAndForget launches a task `fa` as a `Fiber` but doesn't connect it's cancellation system
+
+Both operators can safely capture the running `Fiber` and _guarantee_ not leaking using `bracketCase`.
+An example, writing `raceN` in terms of `racePair`.
+
+```kotlin:ank
+suspend fun <A, B> raceN(ctx: CoroutineContext, fa: suspend () -> A, fb: suspend () -> B): Either<A, B> =
+  bracket(
+    acquire = { racePair(ctx, fa, fb) },
+    use = { it.bimap({ (a, _) -> a }, { (_, b) -> b }) },
+    release = {
+      it.bimap(
+        { (_, fiber) -> fiber.cancel() },
+        { (fiber, _) -> fiber.cancel() }
+      )
+    }
+  )
+```
+
+## Arrow Fx Coroutines, KotlinX Coroutines & Kotlin Standard Library
+ 
+### Demystify Coroutine 
+
+Kotlin's standard library defines a `Coroutine` as an instance of a suspendable computation.
+
+In other words, a `Coroutine` is a compiled `suspend () -> A` program wired to a `Continuation`.
+
+Which can be created by using [`kotlin.coroutines.intrinsics.createCoroutineUnintercepted`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.coroutines.intrinsics/create-coroutine-unintercepted.html).
+
+So let's take a quick look at an example.
+
+```kotlin:ank
+import kotlin.coroutines.intrinsics.createCoroutineUnintercepted
+
+suspend fun one(): Int = 1
+
+val cont: Continuation<Unit> = ::one
+  .createCoroutineUnintercepted(Continuation(EmptyCoroutineContext, ::println))
+
+cont.resume(Unit)
+```
+
+As you can see here above we create a `Coroutine` using `createCoroutineUnintercepted` which returns us `Continuation<Unit>`.
+Strange, you might've expected a `Coroutine` type but a `Coroutine` is represented by `Continuation<Unit>`.
+
+This `typealias Coroutine = Contination<Unit>` will start running every time you call `resume(Unit)`, which allows you to run the suspend program N times.
+
+### Arrow Fx Coroutines & KotlinX Coroutines
+
+Both Arrow Fx Coroutines & KotlinX Coroutines independently offer an implementation for Kotlin's coroutine system.
+
+As explained in the document above, Arrow Fx Coroutines offers a battery-included functional IO with cancellation support.
+Where KotlinX Coroutines offers an implementation that offers light-weight futures with cancellation support.
```

Comment:
Double `offers` on the same line. Maybe join the paragraphs or break them :D

Reply:
"As explained in the document above. Arrow Fx Coroutines offers a battery-included functional IO with cancellation support, while KotlinX Coroutines offers a light-weight futures with cancellation support."

Something like this?

Reply:
"As explained in the document above. Arrow Fx Coroutines offers a battery-included declarative API with cancellation support, while KotlinX Coroutines offers light-weight imperative futures with cancellation support."